grails-app/services/com/unifina/service/CommunityJoinRequestService.groovy

```diff
@@ -34,21 +41,26 @@ class CommunityJoinRequestService {
 		for (Product p : products) {
 			streams.addAll(p.streams)
 		}
+		log.info(String.format("exiting findStreams(): %s", streams))
 		return streams
 	}
 
 	private void sendMessage(CommunityJoinRequest c, String type) {
+		log.info(String.format("entering sendMessage(%s, %s)", c, type))
 		String joinPartStreamID = ethereumService.fetchJoinPartStreamID(c.communityAddress)
+		log.info(String.format("sending message to join part stream id: '%s', community address: '%s'", joinPartStreamID, c.communityAddress))
 		Map<String, Object> msg = new HashMap<>()
 		msg.put("type", type)
 		msg.put("addresses", Arrays.asList(c.memberAddress))
 		sendMessageToStream(joinPartStreamID, msg)
+		log.info("exiting sendMessage")
 	}
 
 	/**
 	 * Sends a message to joinPartStream using the credentials of this Engine node
 	 */
 	private void sendMessageToStream(String joinPartStreamID, HashMap<String, Object> content) {
+		log.info(String.format("entering sendMessageToStream(%s, %s)", joinPartStreamID, content))
 		StreamrClient client = streamrClientService.getInstanceForThisEngineNode()
```

Comment:
Let's log here some stuff about the client (feel free to convert to `String.format`, although IMO this syntax is easier to read)

```
log.debug("sendMessage: StreamrClient state is ${client.getState()}, websocket url: ${client.getOptions().getWebsocketApiUrl()}")
```